src/components/cards/FiatAmountInputCard.tsx

```diff
@@ -41,29 +41,33 @@ const FiatAmountInputCardComponent = ({ wallet, iconUri, inputModalMessage, titl
     const token = tokenId != null ? allTokens[tokenId] : null
     const { denominations: destDenoms } = token != null ? token : wallet.currencyInfo
     const destExchangeMultiplier = destDenoms == null ? '0' : destDenoms[0].multiplier
-
+    
+    // Clean localized fiat amount prior to biggystring ops
+    const fiatAmount = sanitizeDecimalAmount(inputFiatAmount, 2)
     const calculatedNativeCryptoAmount = truncateDecimals(
-      destToFiatRate == null || destToFiatRate === '0' ? '0' : mul(destExchangeMultiplier, div(fiatAmount ?? '0', destToFiatRate, DECIMAL_PRECISION)),
+      destToFiatRate == null || destToFiatRate === '0'
+        ? '0'
+        : mul(destExchangeMultiplier, div(fiatAmount ?? '0', destToFiatRate, DECIMAL_PRECISION)),
       0
     )
 
     setNativeCryptoAmount(calculatedNativeCryptoAmount)
-    if (!zeroString(calculatedNativeCryptoAmount) && !zeroString(fiatAmount)) onAmountChanged(fiatAmount, calculatedNativeCryptoAmount)
-  }, [allTokens, destToFiatRate, fiatAmount, onAmountChanged, tokenId, wallet.currencyInfo])
+    if (!zeroString(calculatedNativeCryptoAmount) && !zeroString(inputFiatAmount)) onAmountChanged(inputFiatAmount, calculatedNativeCryptoAmount)
```

Comment:
This should be reverted so the (now) sanitized value is passed to `zeroString` and `onAmountChanged`. `zeroString` uses biggystring directly as well one of the parents of FiatAmountInputCard passes an `onAmountChanged` that eventually leads to biggystring.